### PR TITLE
fix(button): match experimental spec

### LINKED ISSUE #1928: fix[button]: component audit
## fix[button]: component audit

- [x] Buttons should use tokens for colors, not call out to `ibm-color` variables directly 

- [x] Ghost button needs to have 16 px in these areas as speced; remove the minimum width rule from only our ghost buttons to acheive this (See spec)

image

- [x]  Remove dark boarder around secondary button on hover (there should be no visible boarder based on design spec) 

- [x] Fix small secondary button hover state too, it has the same problem as the large one with the dark boarder on hover

image

### LINKED ISSUE #1935: Button SCSS having hard coded color
## Detailed description
Buttons are having hard coded colors so changing $brand-01 isn't really changing hover color of button
https://github.com/IBM/carbon-components/blob/master/src/components/button/_button.scss#L200
https://github.com/IBM/carbon-components/blob/master/src/components/button/_button.scss#L220

### PR DESCRIPTION
**Pull request by emyarod**
Closes #1928 

Closes #1935

This PR incorporates feedback from the February 2019 button component audit

### CODE REVIEW

**Comment by asudoh** [comment]
@emyarod I see "question" label here - Wondering what the open question is...? Thanks!

**Comment by emyarod** [comment]
@asudoh it was cleared up in the issue thread #1928

**Comment by asudoh** [comment]
Cool, so are you working on the spec update, or is this ready for review?

**Comment by emyarod** [comment]
going to add the latest changes from the updated spec, so this is still a draft PR

File: src/components/button/_button.scss [hunk 1]
```diff
@@ -179,8 +179,6 @@
 @mixin button--x {
   // <button> elements cannot be used as flex containers
   button.#{$prefix}--btn {
-    display: inline-block;
```
Comment (asudoh) [inline src/components/button/_button.scss]: Is this removal intentional...?

File: src/components/button/_button.scss [hunk 2]
```diff
@@ -241,14 +232,11 @@
...
-    @include button-theme--x(
-      transparent,
-      transparent,
-      $interactive-03,
-      $ibm-colors__gray-10,
-      $interactive-03,
-      $ibm-colors__blue-70
-    );
+    @include button-theme--x(transparent, transparent, $interactive-01, $hover-ui, $interactive-01, $active-ui);
+    display: inline-flex;
```
Comment (asudoh) [inline src/components/button/_button.scss:238]: Is this specific to ghost button?

**Review by asudoh — COMMENTED** [review]
LGTM basically, just a set of questions - Thanks @emyarod!

File: src/components/button/_button.scss (hunk 2 above)
Comment (emyarod) [reply src/components/button/_button.scss:238]: yes the position of icon for ghost buttons with icons is slightly different from the other buttons in the spec

File: src/components/button/_button.scss (hunk 1 above)
Comment (emyarod) [reply src/components/button/_button.scss]: I believe the default display value for button elements is `inline-block` right?

Comment (asudoh) [reply src/components/button/_button.scss]: You are right wrt the user-agent stylesheet thing, but I'd guess it's for protecting against application-level overrides. Tracing the code history tells me that the addition was intentional, and I wouldn't remove it unless I have a stronger reason.

File: src/components/button/_button.scss [hunk 3]
```diff
@@ -241,14 +232,11 @@
...
+    align-items: center;
+    min-width: initial;
```
Comment (asudoh) [inline src/components/button/_button.scss]: `initial` seems not supported in IE11: https://developer.mozilla.org/en-US/docs/Web/CSS/initial

**Review by asudoh — COMMENTED** [review]
Thank you for clarifying @emyarod! Some additional comments.

File: src/components/button/_button.scss (hunk 3 above)
Comment (emyarod) [reply src/components/button/_button.scss]: I replaced the `initial` value with `auto`

File: src/components/button/_button.scss (hunk 1 above)
Comment (emyarod) [reply src/components/button/_button.scss]: sounds good, I added that rule back

**Comment by jeanservaas** [comment]
@emyarod  geometry / spacing of the ghost button's looking good now, the only thing that i noticed is that the icon seems slightly high off of vertical center and is not quite aligned with the type. This is happening in both the large and small ghost buttons.

image

**Comment by emyarod** [comment]
@jeanservaas should be resolved now!

**Review by jeanservaas — APPROVED** [review]
LGTM now thanks!

**Review by asudoh — APPROVED** [review]
LGTM 👍 - Thanks @emyarod!
